addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -1519,7 +1519,7 @@ var LinkDialog = Dialog.extend({
             + ((size && size.length) ? (' btn-' + size) : '');
         var isNewWindow = this.$('input[name="is_new_window"]').prop('checked');
 
-        if (url.indexOf('@') >= 0 && url.indexOf('mailto:') < 0) {
+        if (url.indexOf('@') >= 0 && url.indexOf('mailto:') < 0 && !url.match(/^http[s]?/i)) {
```

Comment:
Note: `[]` are useless in that regex :wink: and in fact the whole 's' part is useless as you say 'starts with http or with https'... this is equal to 'starts with http'. If previous condition was not enough, then we should probably use a real email matcher instead of a non-email matcher.